programs/drift-competitions/src/state/competition.rs

```diff
@@ -21,10 +49,123 @@ pub struct Competition {
     pub first_round_expiry_ts: i64,
     pub competition_expiry_ts: i64, // when competition ends, perpetual when == 0
     pub round_duration: u64,
+
+    pub prize_draw: u128,
+    pub winning_draw: u128,
 }
 
 impl Size for Competition {
-    const SIZE: usize = 160 + 8;
+    const SIZE: usize = 200 + 8;
 }
 
-const_assert_eq!(Competition::SIZE, std::mem::size_of::<Competition>() + 8);
\ No newline at end of file
+const_assert_eq!(Competition::SIZE, std::mem::size_of::<Competition>() + 8);
+
+impl Competition {
+    // calculate the end unix timestamp for round_number N
+    pub fn calculate_round_end_ts(&self) -> DriftResult<i64> {
+        self.first_round_expiry_ts
+            .safe_add(self.round_duration.safe_mul(self.round_number)?.cast()?)
+    }
+
+    pub fn validate_round_ready_for_settlement(&self, now: i64) -> CompetitionResult {
+        validate!(
+            self.status == CompetitionRoundStatus::Active,
+            ErrorCode::Default
+        )?;
+        validate!(now >= self.calculate_round_end_ts()?, ErrorCode::Default)?;
```

Comment:
should probably add an error mess in these validates so it's easier to debug in future